homeassistant/components/husqvarna_automower/lawn_mower.py

```diff
@@ -0,0 +1,124 @@
+"""Creates a vacuum entity for the mower."""
+import logging
+
+from aiohttp import ClientResponseError
+
+from homeassistant.components.lawn_mower import (
+    LawnMowerActivity,
+    LawnMowerEntity,
+    LawnMowerEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, ERROR_ACTIVITIES
+from .entity import AutomowerEntity
+
+SUPPORT_STATE_SERVICES = (
+    LawnMowerEntityFeature.DOCK
+    | LawnMowerEntityFeature.PAUSE
+    | LawnMowerEntityFeature.START_MOWING
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up vacuum platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        HusqvarnaAutomowerEntity(coordinator, idx)
+        for idx, ent in enumerate(coordinator.session.data["data"])
+    )
+    entity_platform.current_platform.get()
```

Comment:
Why do we do this?

Reply:
actually not

Reply:
what do you mean?

Reply:
That we don't need it and I have deleted it :)